concordium-consensus/src/Concordium/Queries.hs

```diff
@@ -396,10 +418,10 @@ getNextAccountNonce accountAddress = liftSkovQueryLatest $ do
 -- ** Block indexed
 
 -- |Get the basic info about a particular block.
-getBlockInfo :: BlockHashInput -> MVR finconf (BlockHash, Maybe BlockInfo)
+getBlockInfo :: BlockHashInput -> MVR finconf (Maybe BlockHash, Maybe BlockInfo)
```

Comment:
With all these maybes, can we document that 
(1) in case the second response is Just the first one always is
(2) the first response is Just if and only if the block hash input could be uniquely resolved.

Ideally all of these would have a more precise response type since really only 3 options are possible
Nothing, Nothing
Just, Nothing
Just Just

but that made for more changes so I'm not sure it's worth doing that. But documentation should be added.

Reply:
I think it is worth doing, as I find the multiple Maybe setup harder to read and it affects the whole API here.

Reply:
I have made an attempt at improving this.

Reply:
I think it looks good.